### PR TITLE
Added LastFM Scrobbling. ONLY works if LastFM can resolve entry.title

### PR DESCRIPTION
**Pull request by dbkynd**
Using the entry.title string to query LastFM to get an artist and track to sent back to the scobbling API.
This avoids counting on a similar format since the track data might be <artist> - <track> or <track> ~ <artist>
If there is a more reliable way to get artist and track from say, the youtube_dl lib. That would be the way to go because LastFM doesn't resolve them reliably with a lot of special or non English letters. Though I've been surprised by some tracks that it has done correctly.

It sends a scrobble immediately on track play. Some other apps have scrobbling delayed for up to 10-30 seconds of continuous play for track skips. But I wasn't quite sure how to implement that.

### CODE REVIEW

File: musicbot/scrobble.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+sk = ''
+api_key = ''
+api_secret = ''
+
+class Scrobble(object):
+	def __init__(self, user, passwd, key, secret):
+		global sk
+		global api_key
+		global api_secret
```
Comment (Rapptz) [inline musicbot/scrobble.py:15]: Why are you using globals? Just do `self.sk = ...` and access it via `self`.

File: musicbot/scrobble.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			'username': user,
+			'password': passwd,
+			'api_key': api_key,
+			'api_sig': api_sig
+		}
+		try:
+			sk = self.request(params)['session']['key']
+		except:
+			print('There was an error getting a session key with the LastFM credentials provided.')
+			sys.exit()
```
Comment (Rapptz) [inline musicbot/scrobble.py:31]: You shouldn't exit the entire application because the login failed.

File: musicbot/scrobble.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import hashlib
+import time
+import sys
+
+sk = ''
+api_key = ''
+api_secret = ''
+
+class Scrobble(object):
+	def __init__(self, user, passwd, key, secret):
```
Comment (Rapptz) [inline musicbot/scrobble.py:12]: In Python you can't mix tabs and spaces and it's not recommended to do so inter-file mixing like this. Stick to spaces and don't use tabs.

File: musicbot/scrobble.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		except:
+			print('There was an error scobbling the track ' + entry.title + "'")
+			return
+		
+	def request(self, params):
+		url = "https://ws.audioscrobbler.com/2.0/"
+		params = urllib.parse.urlencode(params).encode()
+		r = urllib.request.Request(url, data=params, headers={"Content-Type": "application/x-www-form-urlencoded"})
+		u = urllib.request.urlopen(r)
+		return json.loads(u.read().decode('utf-8'))
```
Comment (Rapptz) [inline musicbot/scrobble.py:78]: This library is asynchronous. Using e.g. `urllib` is blocking and against the design of the bot. Use `aiohttp` instead and `async def`/`await`.

**Comment by deansheather** [comment]
I thought of this idea as well, but this implementation can only scrobble to one Last.FM account. :disappointed:

**Comment by imayhaveborkedit** [comment]
On top of these issues:
- PRs should be made to the review branch
- This is a feature for after V2 comes out

File: musicbot/scrobble.py (hunk 1 above)
Comment (dbkynd) [inline musicbot/scrobble.py:15]: Literally my first experience with python. Still learning. I appreciate the quick response.

**Pull request closed without merging**